# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsURLHelper.cpp
# Commit: b7fab4b145f7
# Full Hash: b7fab4b145f74d207f421836b32e86625f1759c5
# Author: abhishekmadan30 <148475204+abhishekmadan30@users.noreply.github.com>
# Date: 2024-03-26 21:18:53
# Regressor Bug: 1877419
# File Overlap Count: 1
# Description:
#   Bug 1877419 - Part 1: Added nested stripping of urls to copy without site tracking. r=pbz,necko-reviewers
#   
#   Depends on D204914
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202943
# ==============================================================================

diff -r 78436da20d95 -r b7fab4b145f7 netwerk/base/nsURLHelper.cpp
--- a/netwerk/base/nsURLHelper.cpp	Tue Mar 26 14:42:07 2024 +0000
+++ b/netwerk/base/nsURLHelper.cpp	Tue Mar 26 14:42:07 2024 +0000
@@ -1302,9 +1302,7 @@
   });
 }
 
-namespace {
-
-void SerializeString(const nsCString& aInput, nsAString& aValue) {
+void URLParams::SerializeString(const nsCString& aInput, nsAString& aValue) {
   const unsigned char* p = (const unsigned char*)aInput.get();
   const unsigned char* end = p + aInput.Length();
 
@@ -1325,8 +1323,6 @@
   }
 }
 
-}  // namespace
-
 void URLParams::Serialize(nsAString& aValue, bool aEncode) const {
   aValue.Truncate();
   bool first = true;